### PR TITLE
Create Resend Confirmation reCaptcha connector

### PR DESCRIPTION
**Pull request by sumedhe**
### Proposed changes in this pull request

Related wso2/product-is#6689

### CODE REVIEW

**Review by IsuraD — APPROVED** [review]

File: components/org.wso2.carbon.identity.captcha/src/main/java/org/wso2/carbon/identity/captcha/connector/recaptcha/ResendConfirmationReCaptchaConnector.java [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+    @Override
+    public void init(IdentityGovernanceService identityGovernanceService) {
+
+        this.identityGovernanceService = identityGovernanceService;
+    }
+
+    @Override
+    public int getPriority() {
+
+        return 5;
```
Comment (IsuraD) [inline components/org.wso2.carbon.identity.captcha/src/main/java/org/wso2/carbon/identity/captcha/connector/recaptcha/ResendConfirmationReCaptchaConnector.java:59]: Priority should not be duplicated.

**Review by IsuraD — APPROVED** [review]

File: components/org.wso2.carbon.identity.captcha/src/main/java/org/wso2/carbon/identity/captcha/connector/recaptcha/ResendConfirmationReCaptchaConnector.java [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+import javax.servlet.http.HttpServletResponse;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * This class handle the resend confirmation ReCaptcha.
+ */
+public class ResendConfirmationReCaptchaConnector extends AbstractReCaptchaConnector {
+
+    private static final Log log = LogFactory.getLog(UsernameRecoveryReCaptchaConnector.class);
```
Comment (thanujalk) [inline components/org.wso2.carbon.identity.captcha/src/main/java/org/wso2/carbon/identity/captcha/connector/recaptcha/ResendConfirmationReCaptchaConnector.java:44]: ResendConfirmationReCaptchaConnector

**Review by thanujalk — APPROVED** [review]

File: components/org.wso2.carbon.identity.captcha/src/main/java/org/wso2/carbon/identity/captcha/connector/recaptcha/ResendConfirmationReCaptchaConnector.java [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+
+        if (StringUtils.isBlank(path) || !(CaptchaUtil.isPathAvailable(path, RESEND_CONFIRMATION_URL))) {
+            return false;
+        }
+
+        try {
+            connectorConfigs = CaptchaUtil.getConnectorConfigs(servletRequest, identityGovernanceService,
+                    RESEND_CONFIRMATION_RECAPTCHA_ENABLE);
+        } catch (Exception e) {
+            // Can happen due to invalid tenant/ invalid configuration
```
Comment (DinikaSen) [inline components/org.wso2.carbon.identity.captcha/src/main/java/org/wso2/carbon/identity/captcha/connector/recaptcha/ResendConfirmationReCaptchaConnector.java:76]: Let's end the comments with full stops.

File: components/org.wso2.carbon.identity.captcha/src/main/java/org/wso2/carbon/identity/captcha/internal/CaptchaComponent.java [hunk 4]
```diff
@@ -71,6 +72,10 @@ protected void activate(ComponentContext context) {
             captchaConnector = new PasswordRecoveryReCaptchaConnector();
             captchaConnector.init(CaptchaDataHolder.getInstance().getIdentityGovernanceService());
             CaptchaDataHolder.getInstance().addCaptchaConnector(captchaConnector);
+            // Initialize and register ResendConfirmationReCaptchaConnector
```
Comment (DinikaSen) [inline components/org.wso2.carbon.identity.captcha/src/main/java/org/wso2/carbon/identity/captcha/internal/CaptchaComponent.java:75]: Let's end the comments with full stops.

File: components/org.wso2.carbon.identity.captcha/src/main/java/org/wso2/carbon/identity/captcha/connector/recaptcha/ResendConfirmationReCaptchaConnector.java [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+            throw new CaptchaClientException("reCaptcha response is not available in the request.");
+        }
+        return CaptchaUtil.isValidCaptcha(reCaptchaResponse);
+    }
+
+    @Override
+    public CaptchaPostValidationResponse postValidate(ServletRequest servletRequest, ServletResponse servletResponse)
+            throws CaptchaException {
+
+        //No need post validation.
```
Comment (DinikaSen) [inline components/org.wso2.carbon.identity.captcha/src/main/java/org/wso2/carbon/identity/captcha/connector/recaptcha/ResendConfirmationReCaptchaConnector.java:125]: Let's add a space between slashes and the beginning of the comment.

File: components/org.wso2.carbon.identity.captcha/src/main/java/org/wso2/carbon/identity/captcha/connector/recaptcha/ResendConfirmationReCaptchaConnector.java (hunk 3 above)
Comment (sumedhe) [reply components/org.wso2.carbon.identity.captcha/src/main/java/org/wso2/carbon/identity/captcha/connector/recaptcha/ResendConfirmationReCaptchaConnector.java:76]: Fixed in #474

File: components/org.wso2.carbon.identity.captcha/src/main/java/org/wso2/carbon/identity/captcha/internal/CaptchaComponent.java (hunk 4 above)
Comment (sumedhe) [reply components/org.wso2.carbon.identity.captcha/src/main/java/org/wso2/carbon/identity/captcha/internal/CaptchaComponent.java:75]: Fixed in #474

File: components/org.wso2.carbon.identity.captcha/src/main/java/org/wso2/carbon/identity/captcha/connector/recaptcha/ResendConfirmationReCaptchaConnector.java (hunk 5 above)
Comment (sumedhe) [reply components/org.wso2.carbon.identity.captcha/src/main/java/org/wso2/carbon/identity/captcha/connector/recaptcha/ResendConfirmationReCaptchaConnector.java:125]: Fixed in #474

**Pull request merged**